test/src/e2e_vm_tests/test_programs/should_pass/language/generic_with_other_generic/src/main.sw

```diff
@@ -0,0 +1,23 @@
+script;
+
+struct Generic1<T> {
+    a: T,
+}
+
+struct Generic2<T> {
+    b: Generic1<T>,
+}
+
+enum Generic3<T> {
+  A: T,
+  B: T
+}
+
+enum Generic4<T> {
+  C: Generic3<T>,
+  D: Generic3<T>
+}
+
+fn main() -> u64 {
+  1
```

Comment:
can we try instantiating the structs so we know that the same `T` actually unifies all the way down? I have a fear that it will freshly `copy_types` for each `T` going down...hopefully not though